app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeController.scala

```diff
@@ -186,6 +196,86 @@ class RepresenteeController @Inject() (
     }
   }
 
+  def confirmPerson(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withCapacitorOrPersonalRepresentativeAnswers(request) { (representativeType, journey, answers) =>
+      val backLink = answers.fold(
+        _ => routes.RepresenteeController.enterId(),
+        _ => routes.RepresenteeController.checkYourAnswers()
+      )
+      answers match {
+        case incompleteAnswers @ IncompleteRepresenteeAnswers(Some(_), Some(_), _, None, false) =>
+          Ok(confirmPersonPage(incompleteAnswers, representativeType, journey.isRight, confirmPersonForm, backLink))
+
+        case _ => Redirect(routes.RepresenteeController.checkYourAnswers())
+      }
+    }
+  }
+
+  def confirmPersonSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withCapacitorOrPersonalRepresentativeAnswers(request) { (representativeType, journey, answers) =>
+        val backLink = answers.fold(
+          _ => routes.RepresenteeController.enterId(),
+          _ => routes.RepresenteeController.checkYourAnswers()
+        )
+        def handleForm(incompleteRepresenteeAnswers: IncompleteRepresenteeAnswers): Future[Result] =
+          confirmPersonForm
+            .bindFromRequest()
+            .fold(
+              formWithErrors =>
+                BadRequest(
+                  confirmPersonPage(
+                    incompleteRepresenteeAnswers,
+                    representativeType,
+                    journey.isRight,
+                    formWithErrors,
+                    backLink
+                  )
+                ),
+              isYes =>
+                if (isYes) {
+                  updateDraftReturnAndSession(
+                    incompleteRepresenteeAnswers.copy(hasConfirmedContactDetails = true),
```

Comment:
they haven't confirmed the contact details here, that's the next page - they've confirmed the person

Reply:
Done.